server/src/main/java/org/opensearch/index/SegmentReplicationStatsTracker.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index;
+
+import org.opensearch.index.shard.IndexShard;
+import org.opensearch.index.shard.ShardId;
+import org.opensearch.indices.IndicesService;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Tracker responsible for computing SegmentReplicationStats.
+ *
+ * @opensearch.internal
+ */
+public class SegmentReplicationStatsTracker {
```

Comment:
nit: Can this class be removed with `getStats()` moved to `SegmentReplicationPressureService` as I don't see this class maintaining/tracking segrep stats. Please ignore if there is any pending future work which demands this class's existence.

Reply:
Implemented with this commit - https://github.com/opensearch-project/OpenSearch/pull/6563/commits/473cc02ca701974662cb38edb2ae85959ea1be00.  I've opened that up in draft and will open it for review once this lands.